core/cache/lxdprofilewatcher.go

```diff
@@ -133,22 +131,23 @@ func (w *MachineAppLXDProfileWatcher) init(machine *Machine) error {
 			info.charmProfile = lxdProfile
 		}
 
-		applications[appName] = info
+		w.applications[appName] = info
 	}
-
-	w.applications = applications
-	w.machineId = machine.Id()
 	return nil
 }
 
 // applicationCharmURLChange sends a notification if what is saved for its
 // charm lxdprofile changes.  No notification is sent if the profile pointer
 // begins and ends as nil.
 func (w *MachineAppLXDProfileWatcher) applicationCharmURLChange(topic string, value interface{}) {
-	w.notifyWatcherBase.mu.Lock()
+	// We don't want to respnsd to any events until we have been fully initialized.
```

Comment:
"respond", here and in the other new select statements.

Reply:
Oops. Will fix typo and squash commit.